### PR TITLE
all: normalize the aws account identifiers

### PR DESCRIPTION
**Pull request by manelmontilla**
This PR normalizes the AWS account identifiers to the short form, e.g.: 123456789111, that it gets from the asset inventory before scanning.

### CODE REVIEW

File: graph_altimeter/scan/__init__.py [hunk 1]
```diff
@@ -155,3 +161,26 @@ def remove_scan_files(temp_dir, scan_id):
...
+    """Returned when normalized an AWS account identifier if the format is not
+    recognized."""
+
+    def __init__(self, AWS_account):
+        super().__init__(
+            f'the format of the AWS_account: {AWS_account} is invalid'
+        )
+
+
+def normalize_account_id(aws_account):
```
Comment (roi-martin-adv) [inline graph_altimeter/scan/__init__.py]: s/normalize_account_id/normalize_aws_account/ ?

Otherwise, it is hard to know that this refers to AWS from the package perspective.

If you change it, remember to modify test names and comments in tests/test_normalize_account_id.py

File: graph_altimeter/scan/__init__.py [hunk 2]
```diff
@@ -155,3 +161,26 @@ def remove_scan_files(temp_dir, scan_id):
...
+    """if the input is an AWS account arn, it returns the correspondent
```
Comment (roi-martin-adv) [inline graph_altimeter/scan/__init__.py]: s/correspondent/corresponding/ ?

File: graph_altimeter/scan/__init__.py [hunk 3]
```diff
@@ -155,3 +161,26 @@ def remove_scan_files(temp_dir, scan_id):
...
+    account id. If the input is already an account id it just returns it.
+    Otherwise it raises a InvalidAWSAccount identifier exception.
```
Comment (roi-martin-adv) [inline graph_altimeter/scan/__init__.py]: s/a/an/
s/identifier// ?

File: graph_altimeter/scan/__init__.py [hunk 4]
```diff
@@ -34,12 +35,17 @@
 logger = logging.getLogger(__name__)
 logger.addHandler(logging.NullHandler())
 
+arn_aws_account_re = re.compile('^arn:aws:iam::([0-9]{12}):root$')
```
Comment (roi-martin-adv) [inline graph_altimeter/scan/__init__.py]: s/arn_aws_account_re/aws_account_arn_re/

So it is consistent with `aws_account_id_re`.

File: graph_altimeter/scan/__init__.py [hunk 5]
```diff
@@ -155,3 +161,26 @@ def remove_scan_files(temp_dir, scan_id):
         shutil.rmtree(scan_path)
     except FileNotFoundError:
         logger.debug("temp directory %s not found", scan_path)
+
+
+class InvalidAWSAccount(Exception):
+    """Returned when normalized an AWS account identifier if the format is not
+    recognized."""
```
Comment (roi-martin-adv) [inline graph_altimeter/scan/__init__.py]: I feel this is hard to understand. I would reword it to something like:

```python
"""Returned when trying to normalize a malformed AWS account identifier."""
```

File: tests/test_normalize_account_id.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+"""Tests for normalizing AWS accounts."""
```
Comment (roi-martin-adv) [inline tests/test_normalize_account_id.py]: What about this?

```python
"""Tests for AWS account ID normalization."""
```

File: graph_altimeter/scan/__init__.py [hunk 7]
```diff
@@ -155,3 +161,26 @@ def remove_scan_files(temp_dir, scan_id):
...
+
+    def __init__(self, AWS_account):
```
Comment (roi-martin-adv) [inline graph_altimeter/scan/__init__.py]: s/AWS_account/aws_account/ ?

I feel mixing cases in parameters is not very pythonish.

**Review by roi-martin-adv — CHANGES_REQUESTED** [review]

File: graph_altimeter/scan/__init__.py (hunk 2 above)
Comment (manelmontilla) [reply graph_altimeter/scan/__init__.py]: Absolutely correspondent it's wrong in this context

File: graph_altimeter/scan/__init__.py (hunk 5 above)
Comment (manelmontilla) [reply graph_altimeter/scan/__init__.py]: 👍

File: graph_altimeter/scan/__init__.py (hunk 7 above)
Comment (manelmontilla) [reply graph_altimeter/scan/__init__.py]: agree it isn't.

**Review by jroimartin — APPROVED** [review]
LGTM

**Review by roi-martin-adv — APPROVED** [review]
